### PR TITLE
simple example running locally with movie lens

### PR DESCRIPTION
**Pull request by db7**

### CODE REVIEW

File: cmd/movie-lens-local/main.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+func main() {
+	var (
+		model      = make(map[string]*cofire.Entry)
+		sgd        = cofire.NewSGD(*gamma, *lambda)
+		ratings    = readRatings(*input)
+		train      = ratings[:len(ratings)**sample/100]
+		test       = ratings[len(ratings)**sample/100:]
+		trainError = cofire.NewErrorValidator()
```
Comment (heltonmarx) [inline cmd/movie-lens-local/main.go:67]: the `NewErrorValidator` is not included on `cofire` currently, right ?

Comment (db7) [reply cmd/movie-lens-local/main.go:67]: True! I rebased this PR after PR #3. So better committing #3 first.

File: rating.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package cofire
+
+import (
+	"math/rand"
+)
+
+// Shuffle randomly shuffles a slice of ratings.
+func Shuffle(ratings []Rating) {
+	for i := range ratings {
```
Comment (frairon) [inline rating.go]:
```
:P
rand.Shuffle(len(ratings), func(i, j int){
ratings[i], ratings[j] = ratings[j], ratings[i]
})
```

**Review by frairon — DISMISSED** [review]

File: rating.go (hunk 2 above)
Comment (db7) [reply rating.go]: Oh, I never realized rand had such function. I will use it. Thanks

Comment (db7) [reply rating.go]: Done.

**Review by frairon — APPROVED** [review]

**Pull request merged**